### ISSUE TITLE
Using a model with float input types causes space issue

### ISSUE THREAD

**Issue by lamarvwoods** [microsoft/onnxruntime#11541]
Hey there,
I've been working from your object detection mobile/ios example. When I tried using my own ort model instead of the default one inside the project I get an error: "not enough space: expected 602112, got 150528".
I expected this to be caused by the fact that I changed the element type of the inputTensor to float(your example had it set as uInt8).
My problem is I can't figure out how to give enough space.
The reason I switched to float is my model takes inputs as a float array not a uInt8 array. 
What should I do?

**Comment by RyanUnderhill** [comment]
I'm assuming you're calling CreateTensorWithDataAsOrtValue to create the OrtValue tensor?

The size of the buffer you're passing in needs to be 4x larger (as float is 4 bytes vs uint8 is 1). 602112 = 150528*4

If that isn't the answer, please provide more context for the code you're using

**Comment by lamarvwoods** [comment]
Yes I am calling CreateTensorWithDataAsOrtValue. 

I luckily had figured out that the size needed to be 4x larger but couldn't figure out where to change the value. 
When I multiplied the p_data_byte_count by 4, my app crashed. It crashes on the line that I call run() on my ORTSession. Any thoughts on what I should do?

**Comment by RyanUnderhill** [comment]
You need to ensure the size of the buffer pointed to by p_data is actually the size you're saying it is. Setting p_data_byte_count higher is just lying to onnxruntime and will crash when it tries to access memory that is not there.

You need to buffer you're allocating for p_data to be the correct size. Can you share some code context of how you allocate the buffer and pass it to onnxruntime?

**Comment by lamarvwoods** [comment]
This is almost exactly the code I'm using: https://github.com/microsoft/onnxruntime-inference-examples/blob/main/mobile/examples/object_detection/ios/ORTObjectDetection/ModelHandler.swift
The changes to the code are:

1. I'm using an ort file built from one of my team's onnx models
2. The model has a different order for input  shape. So instead of supplying 1, 300, 300, 3 - I send 1, 3, 244, 244.
3. On line 146 I changed the ORTTensorElementDataType.uInt8 to ORTTensorElementDataType.float because my model takes float32[1,3,224,224] instead of the uint8[1,300,300,3] that the example model does

Does this help?

**Comment by RyanUnderhill** [comment]
Yeah, this won't work without a bunch more changes. You can see the swift code is getting the scaledPixelBuffer from the preprocess method which isn't creating a buffer of float type. You'd need to rewrite it to use floats vs bytes.

**Comment by lamarvwoods** [comment]
Sorry I'm a bit of a novice in these matters. Could you tell me what changes would need to be made?
I keep trying to multiply certain values by 4 but am getting nowhere

**Comment by RyanUnderhill** [comment]
Unfortunately it's a non trivial change and I'm not familiar with Swift. I'd suggest posting it to a Swift community and saying there what you're trying to change, as it's out of scope of an Onnxruntime issue.
